distribution/lib/Standard/Table/0.0.0-dev/src/Internal/In_Memory_Column_Implementation.enso

```diff
@@ -0,0 +1,987 @@
+            False ->
+                apply_unary_operation this_column UnaryRoundOperation.FLOOR_INSTANCE
+
+    coalesce this_column (values : Any | Vector Any) -> Column =
+        vec = Vector.unify_vector_or_element values
+        new_name = naming_helper.function_name "coalesce" [this_column]+vec
+        result = if vec.is_empty then this_column else
+            vec.fold this_column acc-> v-> acc.fill_nothing v
+        result.rename new_name
+
+    min this_column (values : Any | Vector Any) -> Column = Value_Type_Helpers.check_multi_argument_comparable_op this_column values <|
+        fallback a b = if a.is_nothing then b else
+            if b.is_nothing then a else
+                if b < a then b else a
+        _run_vectorized_many_op this_column "min" fallback values
+
+    max this_column (values : Any | Vector Any) -> Column = Value_Type_Helpers.check_multi_argument_comparable_op this_column values <|
+        fallback a b = if a.is_nothing then b else
+            if b.is_nothing then a else
+                if b > a then b else a
+        _run_vectorized_many_op this_column "max" fallback values
+
+    is_nothing_implementation this_column -> Column =
+        new_name = naming_helper.concat [naming_helper.to_expression_text this_column, "is Nothing"]
+        apply_unary_operation this_column IsNothingOperation.INSTANCE new_name
+
+    is_nan this_column -> Column = Value_Type.expect_numeric this_column <|
+        In_Memory_Column_Implementation.internal_is_nan this_column
+
+    ## TODO
+    ## PRIVATE
+    internal_is_nan : Column
+    internal_is_nan this_column =
+        new_name = naming_helper.function_name "is_nan" [this_column]
+        fallback x = case x of
+            _ : Float   -> x.is_nan
+            _           -> False
+        apply_unary_operation this_column DoubleIsOperation.IS_NAN new_name if_unsupported=(apply_unary_map this_column new_name fallback Value_Type.Boolean)
+
+    is_infinite this_column -> Column = Value_Type.expect_numeric this_column <|
+        new_name = naming_helper.function_name "is_infinite" [this_column]
+        fallback x = case x of
+            _ : Float   -> x.is_infinite
+            _           -> False
+        apply_unary_operation this_column DoubleIsOperation.IS_INFINITE new_name if_unsupported=(apply_unary_map this_column new_name fallback Value_Type.Boolean)
+
+    is_finite this_column -> Column = Value_Type.expect_numeric this_column <|
+        new_name = naming_helper.function_name "is_finite" [this_column]
+        fallback x = case x of
+            _ : Float   -> x.is_finite
+            _           -> True
+        apply_unary_operation this_column DoubleIsOperation.IS_FINITE new_name if_unsupported=(apply_unary_map this_column new_name fallback Value_Type.Boolean)
+
+    is_empty this_column -> Column = Value_Type.expect_text this_column <|
+        In_Memory_Column_Implementation.internal_is_empty this_column
+
+    ## TODO
```

Comment:
What is todo?

Reply:
I was planning to move this method out of implementation, but I guess it is okay for it to stay there, it is `private` anyway.